zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -266,3 +290,16 @@ def get_parse_product_request(
         )
         scrapy_request.meta["allow_offsite"] = True
         return scrapy_request
+
+    def _page_params_for_heuristics(
```

Comment:
by the name, I wouldn't expect the page_params to be modified. What do you think about either making a copy, or renaming a method (something like `_modify_page_params_for_heuristics`)?

Reply:
I was initially thinking about making a copy but given that the `page_params` could be arbitrarily nested, using `copy.deepcopy` might take a hit in the performance. 

Renaming it to `_modify_page_params_for_heuristics` seems safer. https://github.com/zytedata/zyte-spider-templates/pull/52/commits/19604aa1476a9876f1b65a473b5ac2aa219ae172